src/extensionConsole/pages/mods/ModsPage.test.tsx

```diff
@@ -0,0 +1,290 @@
+ */
+
+import React from "react";
+import { render, screen } from "@/extensionConsole/testHelpers";
+import ModsPage from "@/extensionConsole/pages/mods/ModsPage";
+import { waitForAct } from "@/testUtils/testHelpers";
+import modsPageSlice from "@/extensionConsole/pages/mods/modsPageSlice";
+import userEvent from "@testing-library/user-event";
+import { appApiMock } from "@/testUtils/appApiMock";
+import { DeploymentsProvider } from "@/extensionConsole/pages/deployments/DeploymentsContext";
+import useAutoDeploy from "@/extensionConsole/pages/deployments/useAutoDeploy";
+import {
+  useAllModDefinitions,
+  useOptionalModDefinition,
+} from "@/modDefinitions/modDefinitionHooks";
+import {
+  errorToAsyncCacheState,
+  loadingAsyncCacheStateFactory,
+  valueToAsyncCacheState,
+} from "@/utils/asyncStateUtils";
+import { waitFor } from "@testing-library/react";
+import { API_PATHS } from "@/data/service/urlPaths";
+import { MODS_PAGE_TABS } from "@/extensionConsole/pages/mods/ModsPageSidebar";
+import { modDefinitionFactory } from "@/testUtils/factories/modDefinitionFactories";
+import { modMetadataFactory } from "@/testUtils/factories/modComponentFactories";
+import { actions as modComponentActions } from "@/store/modComponents/modComponentSlice";
+import useOnboarding from "@/extensionConsole/pages/mods/onboardingView/useOnboarding";
+
+jest.mock("@/modDefinitions/modDefinitionHooks");
+const useAllModDefinitionsMock = jest.mocked(useAllModDefinitions);
+const useOptionalModDefinitionMock = jest.mocked(useOptionalModDefinition);
+
+jest.mock("@/extensionConsole/pages/deployments/useAutoDeploy");
+const useAutoDeployMock = jest.mocked(useAutoDeploy);
+
+jest.mock("@/extensionConsole/pages/mods/onboardingView/useOnboarding");
+const useOnboardingMock = jest.mocked(useOnboarding);
+
+describe("ModsPage", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useAllModDefinitionsMock.mockReturnValue(valueToAsyncCacheState([]));
+    useOptionalModDefinitionMock.mockReturnValue(
+      valueToAsyncCacheState(undefined),
+    );
+    useAutoDeployMock.mockReturnValue({ isAutoDeploying: false });
+    appApiMock.reset();
+    appApiMock.onGet(API_PATHS.MARKETPLACE_LISTINGS).reply(200, []);
+    appApiMock.onGet(API_PATHS.FEATURE_FLAGS).reply(200, {
+      flags: [],
+    });
+    appApiMock.onGet(API_PATHS.BRICKS).reply(200, []);
+  });
+
+  it("renders loading state when mods are loading", async () => {
+    useAllModDefinitionsMock.mockReturnValue(loadingAsyncCacheStateFactory());
+    useOptionalModDefinitionMock.mockReturnValue(
+      loadingAsyncCacheStateFactory(),
+    );
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(true));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("loader")).toBeInTheDocument();
+    });
```

Comment:
waitFor isn't doing anything (tested locally)
```suggestion
      expect(screen.getByTestId("loader")).toBeInTheDocument();
```